src/state.cc

```diff
@@ -69,10 +69,10 @@ Pool State::kDefaultPool("", 0);
 Pool State::kConsolePool("console", 1);
 const Rule State::kPhonyRule("phony");
 
-State::State() : State(NULL) {}
+State::State() : State(std::make_unique<LoggerBasic>()) {}
```

Comment:
Doesn't this change the semantics of the default constructor to change it from having no logger, to having a default logger?